File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        lst=[]
        lengths=[]
        value=0
        if len(s)>0:
            for i in range(len(s)):
                if s[i]!=" ":
                    if s[i] not in lst:
                        lst.append(s[i])
                        lengths.append(len(lst))  
                    elif s[i] in lst and i+1==len(s):
                        lengths.append(len(lst))
                        lst=[]
                        lst.append(s[i])
                    elif s[i] in lst and i+1!=len(s):
                        lengths.append(len(lst))
                        index= lst.index(s[i])
                        lst=lst[index+1:]
                        lst.append(s[i])
                        lengths.append(len(lst))
                elif s.isspace():
                    lengths=[1]
                elif s[i]==" ":
                    lengths.append(len(lst))
                    lst=[]
                    lst.append(s[i])
            value=sorted(lengths)[-1]
        return value

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */


var lengthOfLongestSubstring = function(s) {
    let lst=[];
    let lengths=[];
    let value=0;
    if(s.length>0){
        if(s!=" "){
            for(let i=0; i<s.length; i++){
                if(s[i] != " "){
                    if(lst.includes(s[i])==false){
                        lst.push(s[i]);
                        lengths.push(parseInt(lst.length));
                    }
                    else if(lst.includes(s[i]) && s[i]==s[i-1]){
                        lst=[];
                        lst.push(s[i]);
                        lengths.push(parseInt(lst.length));
                    }
                    else if(lst.includes(s[i])){
                        lst=lst.slice(lst.indexOf(s[i])+1);
                        lengths.push(parseInt(lst.length));
                        lst.push(s[i]);
                    }
                }
                else if(s[i]==" "){
                    lengths.push(parseInt(lst.length));
                    lst=[];
                    lst.push(s[i]);
                };    
            };
        }
        else{
                lengths=[1];
        };     
    }
    else{
        lengths=[0]
    };
    value=Math.max(...lengths);
    return value
};
